### PR TITLE
Corrects incorrect shuffles

### LINKED ISSUE #26098: Misuse of Shuffle()
A lot of code does this: 
```
shuffle(some_list) //A
...
```
And a lot does this:
```
for(var/a in shuffle(some_list)) //B
    ...
```
or this:
```some_list = shuffle(some_list) //C```

B or C is completely fine as ```shuffle()``` returns the new list.
A however, is wrong, as ```shuffle()``` operates on a copy of the list, not the list itself (there's ```shuffle_inplace()``` for that!)

This means that a lot of things that should be being shuffled (ghost role candidates!, late join DAs!) aren't getting shuffled at all.

Either ```shuffle()``` and ```shuffle_inplace()``` need combining, to prevent this issue occuring in future, or we need to teach people that this proc does NOT act in place (probably with a rename?)

I've known this one for quite a while (2+ years) but never got around to writing it up/fixing it, just noticed it again today.

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #26098

### CODE REVIEW

File: code/modules/events/wizard/shuffle.dm [hunk 1]
```diff
@@ -21,8 +21,8 @@
 	if(!mobs)
 		return
 
-	shuffle(moblocs)
```
Comment (AnturK) [inline code/modules/events/wizard/shuffle.dm]: File called shuffle, uses shuffle incorrectly. Top code.

Comment (RemieRichards) [inline code/modules/events/wizard/shuffle.dm]: the irony is almost a physical presence in this realm.

**Comment by xxalpha** [comment]
Does ```shuffle_inplace()``` work?

**Comment by RemieRichards** [comment]
it should, it's logic is the same as ```shuffle()```s, except it acts on the list argument instead of a copy.

**Comment by xxalpha** [comment]
I keep forgetting the real lists are passed and not a copy.

**Pull request merged**